### PR TITLE
refactor(framework): update keras to runners-1.0 branch

### PR DESCRIPTION
**Pull request by larme**
## Checklist:

- [ ] My code follows the bentoml code style, both `make format` and
  `make lint` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by larme** [comment]
will add tests and better docs later

**Comment by parano** [comment]
@larme could you fix the codestyle_check issue?

```
************* Module bentoml.keras
[13](https://github.com/bentoml/BentoML/runs/6603210428?check_suite_focus=true#step:10:14)
E0603:undefined-all-variable [006:46] : Undefined variable name 'to_runner' in __all__
[14](https://github.com/bentoml/BentoML/runs/6603210428?check_suite_focus=true#step:10:15)
W0611:unused-import [004:00] : Unused get_runnable imported from _internal.frameworks.keras
[15](https://github.com/bentoml/BentoML/runs/6603210428?check_suite_focus=true#step:10:16)
************* Module bentoml._internal.frameworks.keras
[16](https://github.com/bentoml/BentoML/runs/6603210428?check_suite_focus=true#step:10:17)
W0613:unused-argument [077:04] load_model: Unused argument 'device_name'
[17](https://github.com/bentoml/BentoML/runs/6603210428?check_suite_focus=true#step:10:18)
E0602:undefined-variable [266:34] get_runnable.KerasRunnable.__init__: Undefined variable 'os'
[18](https://github.com/bentoml/BentoML/runs/6603210428?check_suite_focus=true#step:10:19)
W0611:unused-import [007:00] : Unused Path imported from pathlib
[19](https://github.com/bentoml/BentoML/runs/6603210428?check_suite_focus=true#step:10:20)
W0611:unused-import [010:00] : Unused numpy imported as np
```

**Comment by larme** [comment]
> @larme could you fix the codestyle_check issue?
> 
> ```
> ************* Module bentoml.keras
> [13](https://github.com/bentoml/BentoML/runs/6603210428?check_suite_focus=true#step:10:14)
> E0603:undefined-all-variable [006:46] : Undefined variable name 'to_runner' in __all__
> [14](https://github.com/bentoml/BentoML/runs/6603210428?check_suite_focus=true#step:10:15)
> W0611:unused-import [004:00] : Unused get_runnable imported from _internal.frameworks.keras
> [15](https://github.com/bentoml/BentoML/runs/6603210428?check_suite_focus=true#step:10:16)
> ************* Module bentoml._internal.frameworks.keras
> [16](https://github.com/bentoml/BentoML/runs/6603210428?check_suite_focus=true#step:10:17)
> W0613:unused-argument [077:04] load_model: Unused argument 'device_name'
> [17](https://github.com/bentoml/BentoML/runs/6603210428?check_suite_focus=true#step:10:18)
> E0602:undefined-variable [266:34] get_runnable.KerasRunnable.__init__: Undefined variable 'os'
> [18](https://github.com/bentoml/BentoML/runs/6603210428?check_suite_focus=true#step:10:19)
> W0611:unused-import [007:00] : Unused Path imported from pathlib
> [19](https://github.com/bentoml/BentoML/runs/6603210428?check_suite_focus=true#step:10:20)
> W0611:unused-import [010:00] : Unused numpy imported as np
> ```

done

File: bentoml/_internal/frameworks/keras.py [hunk 1]
```diff
@@ -348,102 +234,101 @@ def KerasSequentialModel() -> keras.models.Model:
...
+            if len(tf.config.list_physical_devices("GPU")) > 0:
+                # In Multi-GPU scenarios, the visible cuda devices will be set for each Runner worker
+                # by the runner's Scheduling Strategy. So that the Runnable implementation only needs
+                # to find the first GPU device visible to current process.
+                self.device_name = "/device:GPU:0"
+            else:
+                self.device_name = "/device:CPU:0"
+                num_threads = int(os.environ["BENTOML_NUM_THREAD"])
+                tf.config.threading.set_inter_op_parallelism_threads(num_threads)
+                tf.config.threading.set_intra_op_parallelism_threads(num_threads)
```
Comment (parano) [inline bentoml/_internal/frameworks/keras.py:276]:
```suggestion
                if "BENTOML_NUM_THREAD" in os.environ:
                    num_threads = int(os.environ["BENTOML_NUM_THREAD"])
                    tf.config.threading.set_inter_op_parallelism_threads(num_threads)
                    tf.config.threading.set_intra_op_parallelism_threads(num_threads)
```

File: bentoml/_internal/frameworks/keras.py [hunk 2]
```diff
@@ -47,141 +45,81 @@
...
 
     """  # noqa
 
-    bentoml_model: Model = model_store.get(tag)
-    if bentoml_model.info.module not in (MODULE_NAME, __name__):
-        raise BentoMLException(
-            f"Model {tag} was saved with module {model.info.module}, failed loading with {MODULE_NAME}."
-        )
+    if not isinstance(bento_model, bentoml.Model):
+        bento_model = bentoml.models.get(bento_model)
```
Comment (parano) [inline bentoml/_internal/frameworks/keras.py:104]:
```suggestion
        bento_model = get(bento_model)
```

**Pull request merged**